### ISSUE TITLE
Modal aria attribute updates

### ISSUE THREAD

**Issue by jgiardino** [patternfly/patternfly-next#1139]
Recently an update was made in the react repo to trap focus in the modal when open: https://github.com/patternfly/patternfly-react/pull/1011

To implement the focus-trapping behavior, `aria-hidden` was added to the wrapping element of the main contents on the page. This should be reflected in core as well.

Updates needed include:

- Update the a11y guide to include a section "Trapping Focus" in [Techniques](https://pf-next.com/accessibility-guide#techniques) with the following information:
    > **Trapping Focus**
    > The recommended interaction pattern for the modal components like the Modal or Popover is to trap focus within the modal element of the component when it becomes visible. For keyboard-only users that use the Tab key to navigate the interface, this means that focus cannot be shifted outside of the modal when using the Tab key. Instead, when focus leaves the last focusable item, it should be placed on the first focusable item of the modal. For screen reader users, the other contents on the page should be hidden from the screen reader. The method we recommend based on the [screen reader / browser combinations we use for testing](https://pf-next.com/accessibility-guide#testing) is to apply `aria-hidden="true"` to the parent wrapping element of the page contents. Note that the modal element of the component must not be a descendent of this element with `aria-hidden="true"` and should be included in a sibling to this element.
- Update the a11y documentation for Modal, Popover, and similar components: 
  - Include a link to the Trapping Focus section in the A11y Guide
  - Include a row in the table of attributes:

    | Attribute | Applies to | Outcome |
    | -- | -- | -- |
    | `aria-hidden="true"` | Parent element containing the page contents | Hides main contents of the page from screen readers. The element with `.pf-c-modal-box` must not be a descendent of the element with `aria-hidden="true"`. **Required**  |

- Update the demos for Modal and AboutModal
  - update the dom so that the modal is separate from the rest of the page contents
  - apply `aria-hidden` to the div that wraps the page contents
  - for example:
    ```
    <body>
      // the div that wraps the main contents of the page
      <div aria-hidden="true">...</div>
      // the div that includes the modal
      <div>...</div>
    </body>
    ```
